test/controllers/CapturePreferenceControllerSpec.scala

```diff
@@ -0,0 +1,188 @@
+  val testValidEmail: String      = "test@example.com"
+  val testInvalidEmail: String    = "invalidEmail"
+  val testYesPreference: String   = "yes"
+  val testNoPreference: String    = "no"
+
+  def target: CapturePreferenceController = new CapturePreferenceController(
+    messagesApi,
+    mockAgentOnlyAuthPredicate,
+    mockConfig
+  )
+
+  "Calling the show action" when {
+
+    "a user is enrolled with a valid enrolment" should {
+
+      lazy val result = target.show(request.withSession(
+        SessionKeys.redirectUrl -> testRedirectUrl
+      ))
+
+      "return 200" in {
+        mockAgentAuthorised()
+        status(result) shouldBe Status.OK
+      }
+
+      "return HTML" in {
+        contentType(result) shouldBe Some("text/html")
+        charset(result) shouldBe Some("utf-8")
+      }
+    }
+
+    "a user is does not have a valid enrolment" should {
+
+      lazy val result = target.show(request.withSession(
+        SessionKeys.redirectUrl -> testRedirectUrl
+      ))
+
+      "return 403" in {
+        mockAgentWithoutEnrolment()
+        status(result) shouldBe Status.FORBIDDEN
+      }
+
+      "return HTML" in {
+        contentType(result) shouldBe Some("text/html")
+        charset(result) shouldBe Some("utf-8")
+      }
+    }
+
+
+    "a user is not logged in" should {
+
+      lazy val result = target.submit(request.withSession(
+        SessionKeys.redirectUrl -> testRedirectUrl
+      ))
+
+      "return 401" in {
+        mockMissingBearerToken()
+        status(result) shouldBe Status.UNAUTHORIZED
+      }
+
+      "return HTML" in {
+        contentType(result) shouldBe Some("text/html")
+        charset(result) shouldBe Some("utf-8")
+      }
+    }
+  }
+
+  "Calling the submit action" when {
+
+    "a user is enrolled with a valid enrolment" when {
+
+      "the user submits after selecting the 'No' option" should {
+
+        lazy val result = target.submit(request
+          .withSession(SessionKeys.redirectUrl -> testRedirectUrl)
+          .withFormUrlEncodedBody("yes_no" -> testNoPreference))
+
+        "redirect to the select client VRN view" in {
+          mockAgentAuthorised()
+          status(result) shouldBe Status.SEE_OTHER
+          redirectLocation(result) shouldBe Some(controllers.agent.routes.SelectClientVrnController.show().url)
+        }
+
+        "add the preference to the session" in {
+          session(result).get(SessionKeys.preference) shouldBe Some(testNoPreference)
+        }
+      }
+
+      "the user submits after selecting the 'Yes' option and an email address" should {
+
+        lazy val result = target.submit(request
+          .withSession(SessionKeys.redirectUrl -> testRedirectUrl)
+          .withFormUrlEncodedBody("yes_no" -> testYesPreference, "email" -> testValidEmail))
+
+        "redirect to the select client VRN view" in {
+          mockAgentAuthorised()
+          status(result) shouldBe Status.SEE_OTHER
```

Comment:
Can this assertion be moved to another test?
It's not best practice to do two in the same test